### PR TITLE
LLM-model-switcher

### PR DESCRIPTION
**Pull request by shreybirmiwal**
Got the UI and the function + ollama to switch models

Stuck on writing the actual function: 

`    def setLLM(self, llm):
        print("Setting LLM...", llm)
`

But will try again tomorrow
Sending PR if anyone knows how to go about this

### CODE REVIEW

File: src/agent/base_agent.py [hunk 1]
```diff
@@ -83,6 +83,8 @@ def __init__(
             max_new_tokens=300,
         )
         self.llm = OpenAI(model="gpt-3.5-turbo-instruct")
```
Comment (kingjulio8238) [inline src/agent/base_agent.py:85]: Lets have llama3 8B as default model

Comment (kingjulio8238) [reply src/agent/base_agent.py:85]: And then for line 80: multimodal model default should be set to Llava or even llava-gradient using ollama with option to change to gpt-4-preview

File: streamlit_app/app.py [hunk 2]
```diff
@@ -88,10 +90,23 @@ def fill_graph(nodes, edges, cypher_query):
...
-query = st.text_input("Ask a question")
 
 
 
+#THIS will fail if the user does not have ollama downloaded - need to fix/find workaround
+models_info = ollama.list()
+available_models = tuple(model["name"] for model in models_info["models"])
+
+if "llama3:8b" in available_models:
+    selected_model = st.selectbox(
```
Comment (kingjulio8238) [inline streamlit_app/app.py:101]: Does this link back to self.llm?

**Review by kingjulio8238 — CHANGES_REQUESTED** [review]

File: streamlit_app/app.py (hunk 2 above)
Comment (shreybirmiwal) [reply streamlit_app/app.py:101]: available models is grabbing from ollama installed models on current device
so no

**Comment by kevinl424** [comment]
> Got the UI and the function + ollama to switch models
> 
> Stuck on writing the actual function:
> 
> ` def setLLM(self, llm): print("Setting LLM...", llm)`
> 
> But will try again tomorrow Sending PR if anyone knows how to go about this

Good stuff here, let me give it a go

**Comment by kingjulio8238** [comment]
All proposed functionalities are included in PR #36 and in release v0.1.2

**Pull request closed without merging**